libs/network/socketserver.c

```diff
@@ -334,13 +338,23 @@ bool socketServerConnectionsIdle(JsNetwork *net) {
       // send data if possible
       JsVar *sendData = jsvObjectGetChild(socket,HTTP_NAME_SEND_DATA,0);
       if (sendData) {
-          if (!socketSendData(net, socket, sckt, &sendData))
+          int sent = socketSendData(net, socket, sckt, &sendData);
+          // FIXME? checking for errors is a bit iffy. With the esp8266 network that returns
+          // varied error codes we'd want to skip SOCKET_ERR_CLOSED and let the recv side deal
+          // with normal closing so we don't miss the tail of what's received, but other drivers
+          // return -1 (which is the same value) for all errors. So we rely on the check ~12 lines
+          // down if(num>0)closeConnectionNow=false instead.
+          if (sent < 0) {
             closeConnectionNow = true;
+            error = sent;
+          }
         jsvObjectSetChild(socket, HTTP_NAME_SEND_DATA, sendData); // socketSendData prob updated sendData
       }
       // only close if we want to close, have no data to send, and aren't receiving data
       if (jsvGetBoolAndUnLock(jsvObjectGetChild(socket,HTTP_NAME_CLOSE,0)) && !sendData && num<=0)
         closeConnectionNow = true;
+      else if (num > 0)
+        closeConnectionNow = false; // guarantee that anything received is processed
```

Comment:
Mhh, sorry, I'm drawing a blank here. You added this comment to line 356-357, which is in the server socket handling and you mention line 525 which is in the client socket handling.